gguf-py/gguf/metadata.py

```diff
@@ -227,6 +236,13 @@ def get_model_id_components(model_id: Optional[str] = None, total_params: int =
                 if part.lower() == "lora":
                     name_parts[i] = "LoRA"
 
+        # Ignore word-based size labels when there is at least a number-based one present
+        if any(c.isdecimal() for n, t in zip(name_parts, name_types) if "size_label" in t for c in n):
+            for n, t in zip(name_parts, name_types):
+                if "size_label" in t:
+                    if all(c.isalpha() for c in n):
+                        t.remove("size_label")
+
```

Comment:
Was not too easy to understand. Got ChatGPT to rewrite it to make sense of it. Seems to checkout but would prefer this to be more grokkable

```python
# Check if there's any size label that includes a number
contains_number_based_size = any(
    any(char.isdecimal() for char in part) 
    for part, types in zip(name_parts, name_types) 
    if "size_label" in types
)

# If a number-based size label is found, process the size labels
if contains_number_based_size:
    for part, types in zip(name_parts, name_types):
        # Only consider entries that have a size label
        if "size_label" in types:
            # Check if the size label is purely alphabetical
            is_alpha_size_label = all(char.isalpha() for char in part)
            # Remove the size label if it's purely alphabetical
            if is_alpha_size_label:
                types.remove("size_label")
```

Reply:
I don't think putting conditions in variables, then checking if that variable is `True` makes anything clearer. It's an unnecessary indirection.

This part especially:

```python3
            # Check if the size label is purely alphabetical
            is_alpha_size_label = all(char.isalpha() for char in part)
            # Remove the size label if it's purely alphabetical
            if is_alpha_size_label:
                types.remove("size_label")
```
And the comments are extremely redundant with the code.

But I can understand why

```python3
any(c.isdecimal() for n, t in zip(name_parts, name_types) if "size_label" in t for c in n)
```
might not be obvious at first.

I first got introduced to that syntax with

https://github.com/ggerganov/llama.cpp/blob/87e397d00bdcedd5cbf6dfda06a7b0f302462728/convert_hf_to_gguf.py#L1559-L1560

The way to see it is that the second `for` is inside the first, while the expression before all the `for`s is as if it was in the innermost `for`.

It's explained in <https://docs.python.org/3.12/reference/expressions.html#generator-expressions> and <https://docs.python.org/3.12/reference/expressions.html#displays-for-lists-sets-and-dictionaries>.